### ISSUE TITLE
Missing icons in notifications

### ISSUE THREAD

**Issue by 0-wiz-0** [dunst-project/dunst#1172]
I'm using dunst from pkgsrc, so there might be a packaging problem - hints welcome!

I followed the blog article https://blog.orhun.dev/introducing-runst/ and started dunst without arguments and no config file, then ran
```
notify-send "runst" "A dead simple notification daemon 🦡" --expire-time 3000                                                                                                                                                                                              
```
A message does pop up, but I get:
```
  WARNING: No icon found in path: 'dialog-information'                                                                                                                                                                                                                     
```
I have a couple icon packages installed:
```
hicolor-icon-theme-0.17                                                                                                                                                                                                              
adwaita-icon-theme-40.1.1                                                                                                                                                                                                                             
oxygen-icons-5.98.0                                                                                                                                                                                                     
gnome-icon-theme-3.12.0                                                                                                                                                                                                    
```
When I copied the default config file coming with dunst 1.9.0 to `~/.config/dunst/dunstrc` I just got more errors:
```
WARNING: Could not find theme Adwaita                                                                                                                                                                                                                                      
WARNING: Could not find theme hicolor                                                                                                                                                                                                                                      
WARNING: Invalid theme index: -1                                                                                                                                                                                                                                           
```
Changing the `icon_path` as the article suggests didn't help either.

Any ideas what could be the problem and how to make dunst find its icons?

**Comment by ShellCode33** [comment]
Hey, you can try

```
enable_recursive_icon_lookup = true
icon_theme = Adwaita  # same name as the folder in /usr/share/icons/
```

**Comment by 0-wiz-0** [comment]
Thank you. I tried that and it didn't help, but your comment did :)
I noticed that dunst really only looks in `/usr/share/icons`- so I fixed that in the pkgsrc package for `icon-lookup.c` and `settings_data.h` and now it works out of the box.

**Comment by iconoclasthero** [comment]
@0-wiz-0 
> Thank you. I tried that and it didn't help, but your comment did :) I noticed that dunst really only looks in `/usr/share/icons`- so I fixed that in the pkgsrc package for `icon-lookup.c` and `settings_data.h` and now it works out of the box.


So wait, you had to recompile it?
Exactly what did you change?

I've got 
```
$ dunst --version
Dunst - A customizable and lightweight notification-daemon 1.12.2 (2025-03-05)
Compiled on 2025-03-25 with the following options:
X11 support: enabled
Wayland support: enabled
SYSCONFDIR set to: /etc/xdg
Compiler flags: -g -std=gnu11 -pedantic -Wall -Wno-overlength-strings -Os -g -O2 -Werror=implicit-function-declaration -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -ffile-prefix-map=/build/dunst-QYusPp/dunst-1.12.2=. -flto=auto -ffat-lto-objects -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -fcf-protection -fdebug-prefix-map=/build/dunst-QYusPp/dunst-1.12.2=/usr/src/dunst-1.12.2-1 -pthread -MMD -MP
Linker flags: -lm -lrt -Wl,-Bsymbolic-functions -flto=auto -ffat-lto-objects -Wl,-z,relro -Wl,-z,now -lgio-2.0 -lgdk_pixbuf-2.0 -lpangocairo-1.0 -lpango-1.0 -lgobject-2.0 -lglib-2.0 -lharfbuzz -lcairo -lwayland-client -lwayland-cursor -lX11 -lXinerama -lXext -lXrandr -lXss

$ neofetch --off ; dunst --startup_notificaion
 
OS: Rhino Linux 2025.2 (server) x86_64 
DE: GNOME 48.0 
WM: Mutter 

WARNING: Could not find theme AdwaitaLegacy
WARNING: Icon 'sound' not found in themes

```

```
$ grep -E icon_theme\|enable_recur\|icon_path dunstrc
    enable_recursive_icon_lookup = true
    icon_theme = "Adwaita"
    # icon_theme = "Adwaita, breeze"
    icon_path = /usr/share/icons/gnome/16x16/status/:/usr/share/icons/gnome/16x16/devices/
$ ll /usr/share/icons/A* -d
drwxr-xr-x -  6 Apr 21:21 /usr/share/icons/Adwaita
```

It looks like it was installed via apt. 

I have the tarball for v1.13.0 so if you can let me know how you fixed it, I'd appreciate it.

**Comment by iconoclasthero** [comment]
Oh, I think I understand where some of this is coming from:
```
$ cat /usr/share/icons/Adwaita/index.theme |head -6
[Icon Theme]
Name=Adwaita
Comment=The Only One
Example=folder
Inherits=AdwaitaLegacy,hicolor
Hidden=true
```

Why I'm getting this when Adwaita is selected and a particular notification is sent is another matter:
`WARNING: Icon 'sound' not found in themes`

**Comment by bynect** [comment]
> find  /usr/share/icons/Adwaita/ -name '*sound*'
>

there is no such icon in adwaita, hence the problem. I found a sound icon in the papirus theme though, maybe that's what you wanted?

**Comment by iconoclasthero** [comment]
> there is no such icon in adwaita, hence the problem. I found a sound icon in the papirus theme though, maybe that's what you wanted?

The warning's coming from the GNOME extension [Media Controls](https://github.com/sakithb/media-controls/blob/main/README.md) (which I now see may be going unmaintained) so I don't know exactly what it's calling for.  I changed over to the [Noia icons](https://store.kde.org/p/1293856) and I think it went away.  (I need to go through the process of starting this as a service and logging errors instead of invoking on the CLI with `&`.)

In scripts I've written I use `notify-send -i audio-x-generic ...` which wasn't a problem.  I can only assume that either the GNOME notification js module was using a different icon set or I just never saw the errors until I switched to dunst yesterday.
